app/src/main/kotlin/github/com/rhacco/dota2androidapp/sources/remote/Dota2OfficialAPIService.kt

```diff
@@ -1,18 +1,38 @@
 package github.com.rhacco.dota2androidapp.sources.remote
 
-import github.com.rhacco.dota2androidapp.api.HeroDataModelResponse
+import github.com.rhacco.dota2androidapp.api.HeroesResponse
+import github.com.rhacco.dota2androidapp.api.RealtimeStatsResponse
+import github.com.rhacco.dota2androidapp.api.TopLiveGamesResponse
 import github.com.rhacco.dota2androidapp.utilities.URLStrings
+import github.com.rhacco.dota2androidapp.utilities.deviceIsOnline
 import io.reactivex.Observable
 import retrofit2.Retrofit
 import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
 import retrofit2.converter.gson.GsonConverterFactory
 import retrofit2.http.GET
 import retrofit2.http.Query
 
+private val sDota2OfficialAPIService by lazy { Dota2OfficialAPIService.create() }
+
+fun getDota2OfficialAPIService(): Dota2OfficialAPIService? =
+        if (deviceIsOnline())
+            sDota2OfficialAPIService
+        else
+            null
```

Comment:
Kotlin, by design, is very averse to null. Rethink this. Avoid null whenever possible and if you do use null, explain why.

Reply:
Thanks, added a comment